cmd/root/results.go

```diff
@@ -27,28 +28,105 @@ func requiredRunes(s string) []rune {
 	return arr
 }
 
-func wpm(events []event) float64 {
+// Remove backspaces from a list of events.
+//
+// Removing backspace events simplifies some calculations,
+// including wpm,
+func removeBackspaces(events []event) []event {
+	enb := []event{}
+	for _, e := range events {
+		if e.typed != "backspace" {
+			enb = append(enb, e)
+		}
+	}
+	return enb
+}
+
+// Calculates the words per minute (wpm) using the events in the list.
+// Also allows the duration to be overridden, which is useful
+// for calculating wpm per second, which is used in the `wpmGraph` function.
+func wpmBase(e []event, raw bool, d time.Duration) float64 {
+	events := removeBackspaces(e)
+	// cannot calculate wpm with less than 2 events
+	if len(events) < 2 {
+		return 0.0
+	}
 	start := events[0].ts
 	end := events[len(events)-1].ts
-	duration := end.Sub(start)
-	mins := float64(duration) / float64(time.Minute)
+	if d == 0 {
+		d = end.Sub(start)
+	}
+	mins := d.Minutes()
 	wordSize := 5.0
-	chars := events[len(events)-1].i + 1
+	chars := len(events)
```

Comment:
This is wrong, but I saved this for later: https://github.com/NicksPatties/sweet/issues/45